src/core/eme/utils/is_session_usable.ts

```diff
@@ -39,14 +39,25 @@ export default function isSessionUsable(
     keyStatuses.push(keyStatus);
   });
 
-  if (keyStatuses.length > 0 &&
-      (
-        !arrayIncludes(keyStatuses, "expired") &&
-        !arrayIncludes(keyStatuses, "internal-error")
-      )
-  ) {
-    log.debug("EME: Reuse loaded session", loadedSession.sessionId);
-    return true;
+  if (keyStatuses.length <= 0) {
+    log.debug("EME: isSessionUsable: MediaKeySession given has an empty keyStatuses",
+              loadedSession);
+    return false;
+  }
+
+  if (arrayIncludes(keyStatuses, "expired")) {
+    log.debug("EME: isSessionUsable: MediaKeySession given has an expired key",
+              loadedSession.sessionId);
+    return false;
   }
-  return false;
+
+  if (arrayIncludes(keyStatuses, "internal-error")) {
```

Comment:
And if the keyStatuses contain at least one "internal-error" **and** "expired" ? Should we consider this case ?

Reply:
You mean that we should alert that both cases coexist when that happens?

I think the current solution is good enough and going overzealous might be harder to maintain if we bring evolution here (though it's really minimal).
The current log messages are still right, they just don't give all the information for a case we're not sure to ever encounter (and if someone does, the user might just have to resolve the first problem, retry, then fix the second).

Reply:
Ok, i understand your point of view.